BrainPortal/app/models/boutiques_portal_task.rb

```diff
@@ -304,8 +310,17 @@ def final_task_list #:nodoc:
         tsk
       end
 
-      original_userfiles_ids = self.params[:interface_userfile_ids].dup
+
```

Comment:
This entire block of changes (lines 313-320) is REALLY hard to read. You use the long ruby expression

```ruby
self.invoke_params[descriptor.sole_mandatory_file_input.id]
```

three times, why don't you extract it in a variable first? Also, why not leave the original line (line 307 in the original code)

```ruby
  original_userfiles_ids = self.params[:interface_userfile_ids].dup
```

exactly as it was, and THEN crush the value of original_userfiles_ids if needed? That would simplify the if/else block